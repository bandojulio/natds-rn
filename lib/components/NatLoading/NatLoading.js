import React from 'react';
import PropTypes from 'prop-types';
import { ActivityIndicator, Modal } from 'react-native';

import { colors } from '../../assets/theme/colors/colors';

import { LoadingView, styles } from './NatLoading.styles';

export const NatLoading = ({ size, color, overlay }) => (
  overlay ? (
    <Modal
    transparent
    overFullScreen
    >
      <LoadingView overlay={overlay}>
      <ActivityIndicator
      style={[styles.container]}
      size={size}
      color={color} />
      </LoadingView>
    </Modal>
  )
  :
  (
    <Modal
    transparent
    overFullScreen
    >
    <ActivityIndicator
      style={[styles.container]}
      size={size}
      color={color} />
    </Modal>
  )
);

NatLoading.propTypes = {
  size: PropTypes.string,
  color: PropTypes.string,
};

NatLoading.defaultProps = {
  size: 'large',
  color: colors.primary.yellow,
};
