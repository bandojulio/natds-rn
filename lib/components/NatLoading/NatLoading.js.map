{"version":3,"sources":["NatLoading.js"],"names":["NatLoading","colors","primary","orange"],"mappings":"iJAAA,4B;AACA;;AAEA;AACA,+C;;AAEO,GAAMA,+BAAa,QAAbA,WAAa;AACxB,8BAAC,0BAAD;AACE,8BAAC,8BAAD,EAAmB,KAAK,OAAxB,CAAgC,MAAOC,eAAOC,OAAP,CAAeC,MAAtD,gDADF,CADwB,GAAnB","file":"NatLoading.js","sourcesContent":["import React from 'react';\nimport { ActivityIndicator } from 'react-native';\n\nimport { colors } from '../../assets/theme/colors/colors';\nimport { LoadingWrapper } from './NatLoading.styles';\n\nexport const NatLoading = () => (\n  <LoadingWrapper>\n    <ActivityIndicator size=\"large\" color={colors.primary.orange} />\n  </LoadingWrapper>\n);\n"]}