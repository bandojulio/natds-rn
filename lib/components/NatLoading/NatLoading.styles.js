import styled from 'styled-components/native';
import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
  },
});

export const LoadingView = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
  background-color: rgba(3, 3, 3, 0.56);
`;
