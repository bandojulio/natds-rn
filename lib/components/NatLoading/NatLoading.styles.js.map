{"version":3,"sources":["NatLoading.styles.js"],"names":["LoadingWrapper","styled","View"],"mappings":"8aAAA,gD;;AAEO,GAAMA,uCAAiBC,iBAAOC,IAAxB,iBAAN","file":"NatLoading.styles.js","sourcesContent":["import styled from 'styled-components/native';\n\nexport const LoadingWrapper = styled.View`\n  justify-content: center;\n  align-items: center;\n  position: absolute;\n  right: 0;\n  left: 0;\n  top: 0;\n  bottom: 0;\n  z-index: 10;\n`;\n"]}