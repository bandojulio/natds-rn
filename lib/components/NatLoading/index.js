Object.defineProperty(exports,"__esModule",{value:true});var _NatLoading=require('./NatLoading');Object.defineProperty(exports,'NatLoading',{enumerable:true,get:function get(){return _NatLoading.NatLoading;}});
//# sourceMappingURL=index.js.map