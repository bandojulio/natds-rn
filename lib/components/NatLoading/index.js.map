{"version":3,"sources":["index.js"],"names":["NatLoading"],"mappings":"mMAASA,U","file":"index.js","sourcesContent":["export { NatLoading } from './NatLoading';\n"]}