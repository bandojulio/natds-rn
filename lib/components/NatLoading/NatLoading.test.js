import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';

import { NatLoading } from './NatLoading';
import { LoadingView } from './NatLoading.styles';


describe('Container component', () => {
  it('Should render loading component correctly', () => {
    const container = renderer.create(
      <NatLoading/>,
    ).toJSON();
    expect(container).toMatchSnapshot();
  });

  it('Should render LoadingView style', () => {
    const style = renderer.create(
      <LoadingView overlay/>
    ).toJSON();
    expect(style).toHaveStyleRule('flex');
    expect(style).toHaveStyleRule('justify-content', 'center');
    expect(style).toHaveStyleRule('align-items', 'center');
    expect(style).toHaveStyleRule('background-color', 'rgba(3, 3, 3, 0.56)');
  });
});
