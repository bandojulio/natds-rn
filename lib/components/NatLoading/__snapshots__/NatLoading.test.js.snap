// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component Should render loading component correctly 1`] = `
<Modal
  hardwareAccelerated={false}
  overFullScreen={true}
  transparent={true}
  visible={true}
>
  <ActivityIndicator
    animating={true}
    color="#F4AB34"
    hidesWhenStopped={true}
    size="large"
    style={
      Array [
        Object {
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  />
</Modal>
`;
